pandas/tests/series/indexing/test_setitem.py

```diff
@@ -1444,10 +1443,6 @@ def obj(self):
             np.float32,
             False,
             marks=pytest.mark.xfail(
-                (
-                    not np_version_gte1p24
-                    or (np_version_gte1p24 and np._get_promotion_state() != "weak")
-                ),
```

Comment:
I think we need to keep just the `not np_version_gte1p24` ? So it is still only skipped for older numpy versions, otherwise we get a xpassed error for recent numpy

Reply:
Ah, not but we do set `NPY_PROMOTION_STATE=legacy` when running the tests on CI, so then indeed we can always skip. However, for local usage, it would be nice to still have this passing if you don't have that env variable set. We could also change `np._get_promotion_state() != "weak"` to something like `os.environ.get("NPY_PROMOTION_STATE", "weak") != "weak"` ?

Reply:
Yeah, that's a good idea. Changed.

Reply:
Thanks! It appears this is actually also needed for our own CI, but apparently we only set that env variable on ubuntu (so that's the reason the mac/windows ones were failing)

Reply:
OK, apparently this is also failing on WASM. Maybe short term you could add a `or WASM` to the xfail condition? (with a `from pandas.compat import WASM`)

Reply:
Done! Thanks for the help on this!